### DISCUSSION TITLE
AI and surrender

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#8289] [General]
Hello everyone,

I have not yet tested the recently added AI surrender feature in fheroes2.
And I think the AI never surrenders in Heroes 2.

However, it shouldn't be "automatic" in fheroes2.
The human player should always accept or reject the AI's surrender.

Remember, a game is primarily designed to entertain the human player. :)

**Comment by zenseii** [comment]
Hi, @LeHerosInconnu. I think this would be a very unfair advantage of the human player. Most definitely you would never accept the surrender of an AI hero if they have the statesman's quill and potentially some diplomacy.

In addition, we would have to make this not happen for Human vs Human battles and I'm personally against having one set of game rules for playing against the AI and another for the human.

**Comment by oleg-derevenetz** [comment]
> However, it shouldn't be "automatic" in fheroes2.
The human player should always accept or reject the AI's surrender.

I'm agree with @zenseii in this regard, because it's unfair (because in case of human vs human battle there is no such privilege) and human player, obviously, will just always decline the AI surrender to capture enemy artifacts and not let the valuable enemy hero to escape. Allowing AI to surrender makes the game more flexible and interesting just by allowing AI to use the same standard mechanics that human player can use.

> Remember, a game is primarily designed to entertain the human player. :)

Challenge may be entertaining as well.

**Comment by LeHerosInconnu** [comment]
Hello @zenseii and @oleg-derevenetz,

> Hi, @LeHerosInconnu. I think this would be a very unfair advantage of the human player. Most definitely you would never accept the surrender of an AI hero if they have the statesman's quill and potentially some diplomacy.

The AI's automatic surrender will already frustrate the player, and in this case even more.

> In addition, we would have to make this not happen for Human vs Human battles and I'm personally against having one set of game rules for playing against the AI and another for the human.

Human players can always talk and get along with each other.
If it's automatic, there's no room for discussion.

You also have to bear in mind that the opponent is not obliged to accept the surrender, and that the hero proposing the surrender is not obliged to agree to pay the cost requested by the opponent.
And as a last resort, he can always flee (only the creatures are lost).

![Reddition 02](https://github.com/ihhub/fheroes2/assets/43583994/d0db85ba-adf3-476b-8b11-11156084a002)

> > However, it shouldn't be "automatic" in fheroes2.
> > The human player should always accept or reject the AI's surrender.
> 
> I'm agree with @zenseii in this regard, because it's unfair (because in case of human vs human battle there is no such privilege) and human player, obviously, will just always decline the AI surrender to capture enemy artifacts and not let the valuable enemy hero to escape. ...

If the human player doesn't accept the AI's surrender, the AI can still flee.

> > Remember, a game is primarily designed to entertain the human player. :)
> 
> Challenge may be entertaining as well.

If the human player wants a "challenge", he accepts the AI's surrender.
And he can also refuse surrender if it suits him better.
With the AI's automatic surrender, there is no choice, and this leads to frustration, not fun.

**Reply by zenseii** [reply]
@LeHerosInconnu, first of all we need to make it clear that what you are asking for is to change the way Heroes 2's surrender was originally designed. I intentionally checked how surrender originally works in Hot seat games and online games, which btw in the latter you wouldn't have always got the chance to negotiate in.

We can discuss new ways to change up the game and make negotiating a part of the gameplay, but for now we have just taught the AI to use a function that previously was a privelege to the human player, exactly like teaching it to use summon boat, use the market place etc. all of which the Heroes 2 AI didn't now how to use.

**Reply by oleg-derevenetz** [reply]
> Human players can always talk and get along with each other.
> If it's automatic, there's no room for discussion.

Regarding human vs human battles, while there is a possibility to discuss something, there is no obligation to perform any discussions. If player has enough money, he just surrenders, and the opposing player can't do anything about it, no matter how much he wants to. I don't see why there should be any special conditions for AI in this regard.

**Reply by LeHerosInconnu** [reply]
@zenseii and @oleg-derevenetz.

> @LeHerosInconnu, first of all we need to make it clear that what you are asking for is to change the way Heroes 2's surrender was originally designed.

No.
In Heroes 2, I never had the case where the AI hero surrendered.
That's how the game was originally designed.
It's the possibility of AI to surrender that is the change (and I didn't ask for this).

> I intentionally checked how surrender originally works in Hot seat games and online games, which btw in the latter you wouldn't have always got the chance to negotiate in.
> 
> We can discuss new ways to change up the game and make negotiating a part of the gameplay, but for now we have just taught the AI to use a function that previously was a privelege to the human player, exactly like teaching it to use summon boat, use the market place etc. all of which the Heroes 2 AI didn't now how to use.

> > Human players can always talk and get along with each other.
> > If it's automatic, there's no room for discussion.
> 
> Regarding human vs human battles, while there is a possibility to discuss something, there is no obligation to perform any discussions. If player has enough money, he just surrenders, and the opposing player can't do anything about it, no matter how much he wants to. I don't see why there should be any special conditions for AI in this regard.

In Heroes 2, in multiplayer mode, the hero proposes his surrender, and is then offered a cost in gold for his surrender, which he can either accept or decline.

Since the players are both human (unless one of them is using "CheatGPT"), they can always agree on how to proceed in the case of surrender before launching the game.
A kind of "negotiation" could also be directly integrated into fheroes2.
Or the opposing human hero could simply decide whether or not to accept surrender (so before the cost of surrender is proposed).
But that's not the point.

In Heroes 2, I never had the case where the AI hero surrendered.
This is a major change from the original game.
When changes are applied, they must be made in such a way as not to frustrate the human player.

That human players frustrate each other, why not.
There's always a way to discuss it, or quit the game (dig for the Ultimate Frustration). :D

But in the case of a game against AI, the game must first be oriented towards the human player's enjoyment of the game.

**Reply by oleg-derevenetz** [reply]
> Since the players are both human (unless one of them is using "CheatGPT"), they can always agree on how to proceed in the case of surrender before launching the game.
A kind of "negotiation" could also be directly integrated into fheroes2.

This is not how surrender mechanics designed. It does not require any consent from the second player. For example, HotA tournament rules (HoMM3 has the same surrender mechanics as HoMM2) have no requirement to ask the opponent's consent when surrendering a hero. You just surrender if you can pay the appropriate amount of money, and that's all. If such consent was required, then the opponent would always refuse if you have a strong hero with valuable artifacts. The human player will do the same with regard to AI - he will just always refuse. It's not fair and this does not correspond to the spirit of the surrender mechanics in its current form.

> In Heroes 2, I never had the case where the AI hero surrendered.
This is a major change from the original game.
When changes are applied, they must be made in such a way as not to frustrate the human player.

In Heroes 2, you never had the case where AI uses the Dimension Door.
In Heroes 2, you never had the case where AI uses the Summon Boat.
In Heroes 2, you never had the case where AI uses the Marketplace.
All this "frustrates" an inexperienced human player because AI wins against him. So what? This is not an argument.

**Reply by zenseii** [reply]
> But in the case of a game against AI, the game must first be oriented towards the human player's enjoyment of the game.

I'd like to add that to me, a human player, I get more enjoyment now that the AI tries to surrender.

Btw in Heroes 4 the AI knows how to surrender and there you get to decide if your opponent should be allowed to surrender or not, and oh boy is it exploitable xD I always consider the enemy heroes on horses as four-legged gold mines in that game.

**Reply by LeHerosInconnu** [reply]
@zenseii and @oleg-derevenetz.

> > Since the players are both human (unless one of them is using "CheatGPT"), they can always agree on how to proceed in the case of surrender before launching the game.
> > A kind of "negotiation" could also be directly integrated into fheroes2.
> 
> This is not how surrender mechanics designed.

AI heroes never surrender in the Heroes 2 original game.
That's how surrender is designed.

> In Heroes 2, you never had the case where AI uses the Dimension Door.
> In Heroes 2, you never had the case where AI uses the Summon Boat.
> In Heroes 2, you never had the case where AI uses the Marketplace. All this "frustrates" an inexperienced human player because AI wins against him. So what? This is not an argument.

I think AI uses Dimension Door in Heroes 2 (certainly not optimally though).
These changes allow the AI to become more efficient, which is good for the game.
Inexperienced human player can remedy this by choosing a lower difficulty level.

> > But in the case of a game against AI, the game must first be oriented towards the human player's enjoyment of the game.
> 
> I'd like to add that to me, a human player, I get more enjoyment now that the AI tries to surrender.

If you agree to let the AI surrender, you will still have that enjoyment.

With the choice of whether or not to let the AI surrender, both types of player are satisfied, not just one type of player.

**Reply by oleg-derevenetz** [reply]
> These changes allow the AI to become more efficient, which is good for the game.

Mechanism of surrender also allows the AI to become more efficient because AI heroes can no longer be trapped in castles during siege and they may save some troops even on the open field (if they have enough money, of course).

> Inexperienced human player can remedy this by choosing a lower difficulty level.

The difficulty level does not affect the AI's ability to trade or summon a boat.

> If you agree to let the AI surrender, you will still have that enjoyment.

Why not make the AI ask for permission for everything? For instance, to summon a boat, build a building or attack you. If you agree to let the AI to do this, you still have the enjoyment, otherwise you won't be frustrated.

**Reply by LeHerosInconnu** [reply]
@oleg-derevenetz.

> > These changes allow the AI to become more efficient, which is good for the game.
> 
> Mechanism of surrender also allows the AI to become more efficient because AI heroes can no longer be trapped in castles during siege and they may save some troops even on the open field (if they have enough money, of course).

In original game, this is so that the human player doesn't get frustrated when an AI hero escapes during a combat in a castle where the AI hero is in the defending position.
Fun is a big part of the original game.

> > Inexperienced human player can remedy this by choosing a lower difficulty level.
> 
> The difficulty level does not affect the AI's ability to trade or summon a boat.

Inexperienced human player can remedy **his frustration** choosing a lower difficulty level (because the AI will perform less efficiently as a whole).

> > If you agree to let the AI surrender, you will still have that enjoyment.
> 
> Why not make the AI ask for permission for everything? For instance, to summon a boat, build a building or attack you. If you agree to let the AI to do this, you still have the enjoyment, otherwise you won't be frustrated.

Personally, I'm against your proposal. :)

**Comment by Branikolog** [comment]
Hi, @LeHerosInconnu .

I personally agree with @zenseii and @oleg-derevenetz . 
Player has a possibility to surrender whether wants AI that or not. Just in case player has enough money. That's how surrendering works in HoMM: "Do you have enough cash? So you can go then!" 
I consider surrendering in HoMM as a "retreat" mechanics for the rich. :)
So I think it would be fair, that AI would have the same possibility to pay the player he fight and surrender. 
Why not? This brings more challenge, when attacking AI that hides inside a castle. You have to consider the possibility, that AI can run away (same as human player can). He pays you money and runs away. I even proposed to make this "unusual" AI behaviour valid only for harder difficulties (https://github.com/ihhub/fheroes2/discussions/8270#discussioncomment-8150577), so player would have more fun playing against stronger and more tricky enemy, that can retreat not only from common battles but also while defending a castle and bring a few more troubles to the player in future. You know, playing easy and picking up every enemy artifact without difficulty is not much fun, right? ;)

The first time AI surrendered from a siege made me so surprised. That was really unexpected thing in that battle. But I really liked enjoyed it. :D

One of our core principles: everything a human player can do, AI should also be able to. If a player can lock Stone liths by standing on it, preventing AI from entering, then AI can block Stone liths himself too. :D (Although from my point of view this abuse should be disabled for both AI and human.)

**Comment by LeHerosInconnu** [comment]
Hello @zenseii, @oleg-derevenetz and @Branikolog,

I don't understand this mindset of wanting to impose things on players when it would be possible to satisfy a maximum number of players.
AI is there to entertain the player.

**Reply by oleg-derevenetz** [reply]
Hi @LeHerosInconnu in #8297 I introduced the levers that could be used to allow or disable AI surrender or retreat (individually) depending on game type (regular or campaign) and game difficulty level. This could be discussed in #8321.

**Comment by LeHerosInconnu** [comment]
This is planned in my mod: https://github.com/ihhub/fheroes2/discussions/9992.
I close the discussion.
